### PR TITLE
Plonky3 fixed columns

### PR DESCRIPTION
**Pull request by Schaeff**
As #1537 was getting too large I had a go at forking p3 and making more minimal changes in powdr.

### CODE REVIEW

File: backend/src/plonky3/mod.rs [hunk 1]
```diff
@@ -50,4 +63,13 @@ impl<'a, T: FieldElement> Backend<'a, T> for Plonky3Prover<T> {
 
         Ok(self.prove(witness, witgen_callback)?)
     }
+
+    fn export_verification_key(&self, output: &mut dyn io::Write) -> Result<(), Error> {
+        let vk = self
+            .get_verifying_key()
+            .ok_or_else(|| Error::BackendError("No verification key set".into()))?;
+        serde_json::to_writer(output, vk).unwrap();
```
Comment (georgwiese) [inline backend/src/plonky3/mod.rs:71]: Would vote for `bincode`, so we get more realistic estimates of the minimal VK size in bytes!

File: plonky3/src/circuit_builder.rs [hunk 2]
```diff
@@ -250,34 +259,47 @@ impl<'a, T: FieldElement> PowdrCircuit<'a, T> {
...
             |(index, ((_, col_id, row_id), public_value))| {
                 //set decr for each public to be row_id in the first row and decrement by 1 each row
                 let (decr, inv_decr, s, decr_next) = (
-                    local[self.analyzed.commitment_count() + 3 * index],
-                    local[self.analyzed.commitment_count() + 3 * index + 1],
-                    local[self.analyzed.commitment_count() + 3 * index + 2],
-                    next[self.analyzed.commitment_count() + 3 * index],
+                    local[public_offset + 3 * index],
+                    local[public_offset + 3 * index + 1],
+                    local[public_offset + 3 * index + 2],
```
Comment (georgwiese) [inline plonky3/src/circuit_builder.rs:301]: Now we can add one fixed column instead of 3 witness columns, right?

Comment (Schaeff) [reply plonky3/src/circuit_builder.rs:301]: Yes, different pr

File: backend/src/plonky3/mod.rs (hunk 1 above)
Comment (Schaeff) [reply backend/src/plonky3/mod.rs:71]: Makes sense

File: plonky3/src/stark.rs [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+        verify_with_key(
+            &config,
+            verifying_key,
+            &circuit,
+            &mut challenger,
+            &proof,
+            &publics,
+        )
+        .unwrap();
+        Ok(serde_json::to_vec(&proof).unwrap())
```
Comment (georgwiese) [inline plonky3/src/stark.rs:174]: Here we still use JSON. Can be fixed in a follow-up PR though.

**Review by georgwiese — APPROVED** [review]
Nice 🚀

**Pull request merged**